search_action_domain/models/models.py

```diff
@@ -0,0 +1,59 @@
+from odoo import api, http, models
+from odoo.tools.cache import ormcache
+
+
+class Base(models.AbstractModel):
+    _inherit = "base"
+
+    @ormcache("self.env.uid", "model", "action_ids", "cache_key")
+    def _get_cached_action_domain_to_include(self, model, action_ids, cache_key=False):
+        actions = (
+            self.env["ir.actions.server"]
+            .sudo()
+            .with_context(search_action_domain=False)
+            .search(
+                [
+                    ("id", "in", action_ids.split(",")),
+                    ("state", "=", "code"),
+                    ("model_id.model", "=", model),
+                ]
+            )
+        )
+        domain = []
+        for action in actions:
+            res = action.with_context(active_model=model).run()
+            res_domain = res and isinstance(res, dict) and res.get("domain")
+            domain += res_domain if res_domain and isinstance(res_domain, list) else []
+        return domain
+
+    @api.model
+    def _get_action_domain_to_include(self):
+        ctx_dict = self.env.context.get("search_action_domain") or {}
+        if not isinstance(ctx_dict, dict):
+            return []
+        action_ids = ctx_dict.get(self._name) or []
+        if (
+            not action_ids
+            or not isinstance(action_ids, list)
```

Comment:
Remove. There shouldn't be wrong types. And if they are, I consider it's better to fail than slowing down every correct operation.

Same for action IDs

Reply:
Done!